CRM/Civicase/Form/CaseSalesOrderContributionCreate.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+use Civi\Api4\OptionValue;
+use CRM_Certificate_ExtensionUtil as E;
+
+/**
+ * Case sales order contribution create Form controller class.
+ *
+ * @see https://docs.civicrm.org/dev/en/latest/framework/quickform/
+ */
+class CRM_Civicase_Form_CaseSalesOrderContributionCreate extends CRM_Core_Form {
+
+  /**
+   * Case sales order to delete.
+   *
+   * @var int
+   */
+  public $id;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function preProcess() {
+    CRM_Utils_System::setTitle('Create Contribution');
+
+    $this->id = CRM_Utils_Request::retrieve('id', 'Positive', $this);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function buildQuickForm() {
+    $this->addElement('radio', 'to_be_invoiced', '', ts('Enter % to be invoiced ?'),
+      'percent', [
+        'id' => 'invoice_percent',
+      ]);
+    $this->add('text', 'percent_value', '', [
+      'id' => 'percent_value',
+      'placeholder' => 'Percentage to be invoiced',
+      'class' => 'form-control',
+      'min' => 1,
+      'max' => 10,
+    ], FALSE);
+    $this->addElement('radio', 'to_be_invoiced', '', ts('Remaining Balance'),
+      'remain',
+      ['id' => 'invoice_remain']
+    );
+    $this->addRule('to_be_invoiced', ts('Invoice value is required'), 'required');
+
+    $statusOptions = OptionValue::get()
+      ->addSelect('value', 'label')
+      ->addWhere('option_group_id:name', '=', 'case_sales_order_status')
+      ->execute()
+      ->getArrayCopy();
+    array_combine(array_column($statusOptions, 'label'), array_column($statusOptions, 'value'));
+
+    $this->add(
+      'select',
+      'status',
+      ts('Status'),
+      array_merge(
+        ['' => 'Select'],
+        array_combine(
+          array_column($statusOptions, 'value'),
+          array_column($statusOptions, 'label')
+        ),
+      ),
+      TRUE,
+      ['class' => 'form-control']
+    );
+
+    $this->addButtons([
+      [
+        'type' => 'submit',
+        'name' => E::ts('Create Contribution'),
+      ],
+      [
+        'type' => 'cancel',
+        'name' => E::ts('Cancel'),
+        // 'isDefault' => TRUE,
+      ],
+    ]);
+
+    parent::buildQuickForm();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function addRules() {
+    $this->addFormRule([$this, 'formRule']);
+  }
+
+  /**
+   * Form Validation rule.
+   *
+   * This enforces the rule whereby,
+   * user must supply an amount if the
+   * enter percentage smount radio is selected.
```

Comment:
smount -> amount

Reply:
>  The PR gif does not show a successful contributing creation and only shows the form validation, so can add another gif for that (which I as far as I can and in case of successful submission, it should do a redirect to the contribution page with some prefilled values) .

Right chief, I didn't complete the logic in this PR, the core logic will be added in another PR. I wanted this PR to focus on the popup itself